### ISSUE TITLE
UX: better feedback when find_program() is false and user forgot disabler: true

### ISSUE THREAD

**Issue by scivision** [mesonbuild/meson#5815]
If a user does

```meson
x  = find_program('foo', required: false)

test('Foo', x)
``` 
For this use case, they should also set `disabler: true` option. If they don't, Meson hard fails with error below. Would it be better UX to instead inform the user they have used an incorrect Meson type, maybe even that they should use disabler: true?

```
Traceback (most recent call last):
  File "meson/mesonbuild/mesonmain.py", line 129, in run
    return options.run_func(options)
  File "meson/mesonbuild/mtest.py", line 1035, in run
    return th.doit()
  File "meson/mesonbuild/mtest.py", line 792, in doit
    self.run_tests(tests)
  File "meson/mesonbuild/mtest.py", line 944, in run_tests
    self.drain_futures(futures)
  File "meson/mesonbuild/mtest.py", line 959, in drain_futures
    result.result()
  File "/home/box/miniconda3/lib/python3.7/concurrent/futures/_base.py", line 425, in result
    return self.__get_result()
  File "/home/box/miniconda3/lib/python3.7/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "/home/box/miniconda3/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "meson/mesonbuild/mtest.py", line 476, in run
    cmd = self._get_cmd()
  File "meson/mesonbuild/mtest.py", line 456, in _get_cmd
    if self.test.fname[0].endswith('.jar'):
AttributeError: 'NoneType' object has no attribute 'endswith'
```

**Comment by scivision** [comment]
This is the patch that "fixes" this traceback:

```patch
diff --git a/mesonbuild/mtest.py b/mesonbuild/mtest.py
index dc5c9d19..87a31997 100644
--- a/mesonbuild/mtest.py
+++ b/mesonbuild/mtest.py
@@ -453,6 +453,8 @@ class SingleTestRunner:
         self.options = options

     def _get_cmd(self) -> typing.Optional[typing.List[str]]:
+        if self.test.fname[0] is None:
+            return None
         if self.test.fname[0].endswith('.jar'):
             return ['java', '-jar'] + self.test.fname
         elif not self.test.is_cross_built and run_with_mono(self.test.fname[0]):
```

**Comment by xclaesse** [comment]
IMHO, the proper fix in in interpreter.py, method add_test(), if exe is a ExternalProgramHolder you should check exe.found(), and raise an InterpreterException if not.

**Comment by eli-schwartz** [comment]
This was reported again in #10091 and fixed in #10092
